cmd/list/machinepool/cmd.go

```diff
@@ -28,38 +29,49 @@ import (
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var Cmd = &cobra.Command{
-	Use:     "machinepools",
-	Aliases: []string{"machinepool", "machine-pools", "machine-pool"},
-	Short:   "List cluster machine pools",
-	Long:    "List machine pools configured on a cluster.",
-	Example: `  # List all machine pools on a cluster named "mycluster"
-  rosa list machinepools --cluster=mycluster`,
-	Run:  run,
-	Args: cobra.NoArgs,
-}
+const (
+	use     = "machinepools"
+	short   = "List cluster machine pools"
+	long    = "List machine pools configured on a cluster."
+	example = `  # List all machine pools on a cluster named "mycluster"
+  rosa list machinepools --cluster=mycluster`
+)
 
-func init() {
-	ocm.AddClusterFlag(Cmd)
-	output.AddFlag(Cmd)
-}
+var (
+	aliases = []string{"machinepool", "machine-pools", "machine-pool"}
+)
 
-func run(_ *cobra.Command, _ []string) {
-	r := rosa.NewRuntime().WithAWS().WithOCM()
-	defer r.Cleanup()
+func NewListMachinePoolCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   short,
+		Long:    long,
+		Aliases: aliases,
+		Example: example,
+		Args:    cobra.MaximumNArgs(1),
```

Comment:
Why did this change from `cobra.NoArgs`?

Reply:
oh woops, shouldnt have, did it while looking at what i did for describe